balancer/weightedroundrobin/balancer_test.go

```diff
@@ -460,6 +460,65 @@ func (s) TestBalancer_TwoAddresses_OOBThenPerCall(t *testing.T) {
 	checkWeights(ctx, t, srvWeight{srv1, 10}, srvWeight{srv2, 1})
 }
 
+// TestEndpoints_SharedAddress tests the case where two endpoints have the same
+// address. The expected behavior is undefined, however the program should not
+// crash.
+func (s) TestEndpoints_SharedAddress(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
+	defer cancel()
+
+	srv := startServer(t, reportCall)
+	sc := svcConfig(t, perCallConfig)
+	if err := srv.StartClient(grpc.WithDefaultServiceConfig(sc)); err != nil {
+		t.Fatalf("Error starting client: %v", err)
+	}
+
+	endpointsSharedAddress := []resolver.Endpoint{{Addresses: []resolver.Address{{Addr: "same-address"}}}, {Addresses: []resolver.Address{{Addr: "same-address"}}}}
+	srv.R.UpdateState(resolver.State{
+		Endpoints: endpointsSharedAddress,
+	})
+
+	// Make some RPC's and make sure doesn't crash.
+	for i := 0; i < 10; i++ {
+		srv.Client.EmptyCall(ctx, &testpb.Empty{}, grpc.WaitForReady(true))
```

Comment:
Why wait for ready?  It will never be ready as is, right?  So this takes 10 seconds to run?

Reply:
I don't remember why I had it, I think copied from the call option I plumbed in above. Deleted, it's weird though it didn't take 10 seconds to run or fail.